openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java

```diff
@@ -179,8 +179,9 @@ public void initialize(OpenMetadataApplicationConfig config) {
     this.authenticationConfiguration = config.getAuthenticationConfiguration();
     SmtpSettings smtpSettings = config.getSmtpSettings();
     this.isEmailServiceEnabled = smtpSettings != null && smtpSettings.getEnableSmtpServer();
-    this.dao.initializeUsers(config);
+    // Keep this before initializeUsers, else getUpdater() will fail
     SubjectCache.initialize();
+    this.dao.initializeUsers(config);
```

Comment:
this can run into other issues
isn't subjectCache requires user repository to be available
and how does subject cache will have the admins pre-initialized?

Reply:
when we have user migrating from any other SSO to basic auth, during initlization we check populate authMechanism which was null earlier with a randomly generated password, in that case SubjectContext.getInstance will be null